src/sdam/monitor.ts

```diff
@@ -153,6 +157,8 @@ export class Monitor extends TypedEventEmitter<MonitorEvents> {
     }
 
     this.connectOptions = Object.freeze(connectOptions);
+
+    this.mongoLogger = mongoLogger;
```

Comment:
nit, can we put this at the top of the constructor, next to where we attach the server seems like a good spot. The lower end of this function deals with some option reconfiguring that isn't relevant to this step.